Sources/Components/Collection.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  CollectionView.swift
+//  PortalView
+//
+//  Created by Argentino Ducret on 4/4/17.
+//  Copyright © 2017 Guido Marucci Blas. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+public struct CollectionProperties<MessageType> {
+    
+    public var items: [CollectionItemProperties<MessageType>]
+    public var layoutValues: CollectionViewLayoutValues
```

Comment:
I think it would be better not to have this property at all. For now we are not going to support any other type of collection layout so I would put all properties that are currently in `CollectionViewLayoutValues` directly here. Keep in mind that you have to remove any dependency with UIKit.